coffeecard/CoffeeCard.WebApi/Controllers/v2/TicketsController.cs

```diff
@@ -50,5 +47,27 @@ public async Task<ActionResult<List<TicketResponse>>> Get([FromQuery] bool inclu
 
             return Ok(await _ticketService.GetTickets(user, includeUsed));
         }
+
+        /// <summary>
+        /// Uses the ticket on the specified menu item
+        /// </summary>
+        /// <param name="ticketId">The id of the ticket to be used</param>
+        /// <param name="menuItemId">The id of the menu item to use the ticket on</param>
+        /// <returns>The ticket that was used</returns>
+        /// <response code="200">Successful request</response>
+        /// <response code="401">Invalid credentials</response>
+        /// <response code="403">Menu item cannot be redeemed by this ticket</response>
+        /// <response code="404">No ticket on account or menu item not found</response>
+        [ProducesResponseType(typeof(UsedTicketResponse), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(void), StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(MessageResponseDto), StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(MessageResponseDto), StatusCodes.Status404NotFound)]
+        [HttpPost("use/{ticketId:int}/{menuItemId:int}")]
+        public async Task<ActionResult<UsedTicketResponse>> UseTicket(int ticketId, int menuItemId)
```

Comment:
I like your design idea to utilize route/path parameters but I believe this does not read as one would normally expect for REST APIs. You use the ticketId but in fact the TicketService takes a ProductId and then the MenuItem Id. Due to this I think this becomes a bit confusing when have the endpoint like

```
POST tickets/use/{productId}/{menuItemId} (Pull Request)
```

Usually you would expect that we would point to a specific ticketId like
```
POST tickets/{id}/use ..
```
but since we use the productId for user friendlyness, I believe this does not read well.

I'm trying to find a good suggestion how you could make this more inline with REST practices (there's no standard though) which would read like the sentence _Use one ticket for a given product for a given menu item_. I can't think of a good design with paths, so my best suggestion is as we did before

```
POST tickets/use
{
   productId: 1,
   useForMenuItem: 1
}
```

Reply:
Also, if using Route Parameters, then lets use the `FromRoute` annotation for clarity like

```suggetion
public async Task<ActionResult<UsedTicketResponse>> UseTicket([FromRoute] int ticketId, [FromRoute] int menuItemId)
```

Reply:
Addressed in 15a7eec0f68f80a50105d0ffa8e04022c4589783